### PR TITLE
add module for biohansel

### PR DESCRIPTION
**Pull request by rpetit3**
## PR checklist

Closes https://github.com/nf-core/modules/issues/2233 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/biohansel/main.nf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+process BIOHANSEL {
+    tag "$meta.id"
+    label 'process_low'
```
Comment (SPPearce) [inline modules/nf-core/biohansel/main.nf:3]: Is this the correct default for this module? How much resources does it actually use?

File: modules/nf-core/biohansel/main.nf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        'https://depot.galaxyproject.org/singularity/bio_hansel%3A2.6.1--py_0':
+        'quay.io/biocontainers/bio_hansel:2.6.1--py_0' }"
+
+    input:
+    tuple val(meta), path(seqs)
+    val scheme
+    path scheme_metadata
+
+    output:
+    tuple val(meta), path("summary.txt")       , emit: summary
```
Comment (SPPearce) [inline modules/nf-core/biohansel/main.nf:16]: Should these output files have a prefix?

File: modules/nf-core/biohansel/main.nf [hunk 3]
```diff
@@ -0,0 +1,45 @@
+process BIOHANSEL {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::bio_hansel=2.6.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bio_hansel%3A2.6.1--py_0':
```
Comment (SPPearce) [inline modules/nf-core/biohansel/main.nf:7]: This is strange, that `%3` is the unicode for a colon...

File: modules/nf-core/biohansel/main.nf (hunk 1 above)
Comment (rpetit3) [reply modules/nf-core/biohansel/main.nf:3]: Can be multi threaded, so I selected the lowest with multiple cores.

**Review by SPPearce — APPROVED** [review]
I'm happy provided the CI tests are ;)

File: modules/nf-core/biohansel/main.nf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::bio_hansel=2.6.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bio_hansel:2.6.1--py_0':
+        'quay.io/biocontainers/bio_hansel:2.6.1--py_0' }"
+
+    input:
+    tuple val(meta), path(seqs)
+    val scheme
```
Comment (nvnieuwk) [inline modules/nf-core/biohansel/main.nf:12]: You can add this to your meta and supply this in the `args` similar to this [example](https://github.com/nf-core/modules/blob/master/tests/modules/nf-core/gatk4/printsvevidence/nextflow.config). Input fields should be reserved for actual file inputs and all values should be in `meta` according to the `nf-core` guidelines on modules

File: modules/nf-core/biohansel/main.nf [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    tuple val(meta), path("${prefix}-simple-summary.txt"), emit: simple_summary
+    path "versions.yml"                                  , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def scheme_metadata_opt = scheme_metadata ? "--scheme-metadata ${scheme_metadata[0]}" : ""
```
Comment (nvnieuwk) [inline modules/nf-core/biohansel/main.nf:27]: What is the reason for taking the first value in the channel here?

File: modules/nf-core/biohansel/main.nf [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    """
+    hansel \\
+        $args \\
+        $scheme_metadata_opt \\
+        --threads $task.cpus \\
+        --scheme $scheme \\
+        --output-summary ${prefix}-summary.txt \\
+        --output-kmer-results ${prefix}-kmer-results.txt \\
+        --output-simple-summary ${prefix}-simple-summary.txt \\
+        $input_type $seqs
```
Comment (nvnieuwk) [inline modules/nf-core/biohansel/main.nf:38]:
```suggestion
        $input_type \\
        $seqs
```

Small layout change

File: modules/nf-core/biohansel/meta.yml [hunk 7]
```diff
@@ -0,0 +1,58 @@
+name: "biohansel"
+description: write your description here
```
Comment (nvnieuwk) [inline modules/nf-core/biohansel/meta.yml:2]: Please write a description here :)

**Review by nvnieuwk — DISMISSED** [review]

File: modules/nf-core/biohansel/main.nf (hunk 5 above)
Comment (rpetit3) [reply modules/nf-core/biohansel/main.nf:27]: honestly, probably just old habit, will clean it up

File: modules/nf-core/biohansel/meta.yml (hunk 7 above)
Comment (rpetit3) [reply modules/nf-core/biohansel/meta.yml:2]: nice catch, brain wrote it, fingers did not

File: modules/nf-core/biohansel/main.nf (hunk 5 above)
Comment (nvnieuwk) [reply modules/nf-core/biohansel/main.nf:27]: No problem, I was just curious what the reason was :stuck_out_tongue:

File: modules/nf-core/biohansel/meta.yml (hunk 7 above)
Comment (nvnieuwk) [reply modules/nf-core/biohansel/meta.yml:2]: happens to the best of us!

File: modules/nf-core/biohansel/main.nf (hunk 4 above)
Comment (nvnieuwk) [reply modules/nf-core/biohansel/main.nf]: Or you could just supply the raw value in `modules.config` (that would be even easier if you always use the same scheme)

**Pull request merged**